### ISSUE TITLE
si prefix mode when exponent equals 0

### ISSUE THREAD

**Issue by jagerber48** [jagerber48/sciform#174]
With `sciform` we can do

```
print(f'{SciNum(0.1, 0.023):!2rp()}')
# 100(23) m
```

It can be convenient to use this to include SI units

```
print(f'{SciNum(0.1, 0.023):!2rp()}s')
# 100(23) ms
```

However this pattern breaks if the exponent is zero
```
print(f'{SciNum(100, 23):!2rp()}s')
# 100(23)s
```
Note the lack of a space between the unit and the uncertainty. This goes against NIST/BIPM recommendations. This could be solved by mapping an exponent of 0 to `' '` instead of `''` as is done now.

However, this needs to be reconciled with

```
print(f'{SciNum(100, 23):!2ep()}s')
# 100(23)e+02s
```

It's not clear what the right path is here. And, thus far, `sciform` has made the decision to NOT support working directly with SI units since I think that will open up a large can of worms. Perhaps a simple helper function that accepts a `SciNum` or `value` and `uncertainty` and simple unit string, detects if the last character is an SI prefix, and if so, appends the unit string, if not, appends a space and then the unit string.

**Comment by jagerber48** [comment]
Here is where I'm at:

```
print(f'{SciNum(100, 23):!2rp()}s')
# 100(23)s
```

This is fine. There should not be an extra space added, even in prefix mode.

I think custom code is really needed to append units to value or value/uncertainty pairs. 

- If the value/uncertainty pair is a parts per form then units should be forbidden. `123 ppm Hz` doesn't make sense.
- If the value/uncertainty is an SI or IEC prefix then the unit should be appended directly to the resulting string with no space, e.g. `123 k`  becomes `123 kHz` or `123 Ki` becomes `123 KiB`.
- If the value/uncertainty pair is neither parts per, SI, or IEC prefix then the unit should appended with a preceding space so that `123e+03` becomes `123e+03 s` or `123` becomes `123 s`.

A crude way to accomplish this would be to check if `pp` appears in the string. If so, it is parts per and should not allow units. Otherwise check if the last character is alphabetic (or any character than can be a prefix like `μ`) and if so add the unit with no space, if not add the unit with a space.

Such a function could be added as a utility within `sciform`. It might be more natural to include an optional `unit` kwarg in the `Formatter` or `SciNum` and the above logic can be handled within the formatting code.

This is pretty fine for simple units like `s`, `Hz`, `kG`, etc. My hesitation with including code like this is people will want to use it for something like `m/s`. This would work fine. But then they may want it for something like `cm/ms`. Again, this will work if they pass in `m/ms` as their units but now they're having to do manipulation of their value and uncertainty to make it so the denominator makes sense as `ms` AND sciform is doing work to makes sure the numerator is whatever prefix it needs. At this point they should use a package more properly designed for handling units.

**Comment by jagerber48** [comment]
This  code will cover most use cases.

```
import re

from sciform import SciNum


def add_units(number_str: str, unit: str) -> str:
    """Add units to a number string."""
    if re.match(r"[a-zA-Z]", number_str[-1]):
        return f"{number_str}{unit}"
    return f"{number_str} {unit}"


num_str = f"{SciNum(123e6, 0.45e6):rp()}"
print(num_str)
# 123.00(45) M
print(add_units(num_str, 'Hz'))
# 123.00(45) MHz

num_str = f"{SciNum(123e6, 0.45e6):e()}"
print(num_str)
# 1.2300(45)e+08
print(add_units(num_str, 'Hz'))
# 1.2300(45)e+08 Hz

```

For now I will include this as an example/recipe in the documentation and leave it out of the source code.. If there is demand for it to built-in to `sciform` then I will reconsider.

**Comment by jagerber48** [comment]
Closed by #176 May consider integration into `FormattedValue` in the future.
